### PR TITLE
Update listener docs

### PR DESCRIPTION
**Pull request by pedronasser**
Closes https://github.com/iron-io/functions/issues/365

### CODE REVIEW

File: docs/operating/extending.md [hunk 1]
```diff
@@ -2,17 +2,61 @@
 
 IronFunctions is extensible so you can add custom functionality and extend the project without needing to modify the core.
 
+### Replaceable Components
+
+TODO
```
Comment (ccirello) [inline docs/operating/extending.md]: If nothing to write here, perhaps remove this section?

File: docs/operating/extending.md [hunk 2]
```diff
@@ -2,17 +2,61 @@
...
+}
 ```
+
+#### Listener Requirements
+
+To be a valid listener your struct should respect interfaces combined or alone found in the file [listeners.go](/iron-io/functions/blob/master/api/ifaces/listeners.go)
+
+These are all available listeners:
+
+- AppCreateListener
```
Comment (ccirello) [inline docs/operating/extending.md]: What do you think of adding a diagram or perhaps longe explanation for these interfaces? So people understand better exactly which part of the request cycle is being affected.

File: docs/operating/extending.md (hunk 1 above)
Comment (pedronasser) [inline docs/operating/extending.md]: Agreed, I'll remove it

File: docs/operating/extending.md (hunk 2 above)
Comment (pedronasser) [inline docs/operating/extending.md]: Okay. I'll describe each better

**Pull request merged**